pyaedt/edb.py

```diff
@@ -3356,3 +3358,145 @@ def calculate_initial_extent(self, expansion_factor):
         max_width = max_width * expansion_factor
         self.logger.info("The W factor is {}, The initial extent = {:e}".format(expansion_factor, max_width))
         return max_width
+
+    @pyaedt_function_handler()
+    def copy_zones(self, working_directory=None):
+        """Copy multizone EDB project to one new edb per zone.
+
+        Parameters
+        ----------
+        working_directory : str
+            Directory path where all EDB project are copied, if empty will use the current EDB project.
+
+        Returns
+        -------
+           dict[str](EDB PolygonData)
+           Return a dictionary with edb path as key and EDB polygon Data defining the region.
+
+        """
+        if working_directory:
+            if not os.path.isdir(working_directory):
+                os.mkdir(working_directory)
+            else:
+                shutil.rmtree(working_directory)
+                os.mkdir(working_directory)
+        else:
+            working_directory = os.path.dirname(self.edbpath)
+        zone_primitives = list(self.active_layout.GetZonePrimitives())
+        edb_zones = {}
+        if not self.setups:
+            self.siwave.add_siwave_syz_analysis()
+            self.save_edb()
+        for zone in zone_primitives:
+            edb_zone_path = os.path.join(
+                working_directory, "{}_{}".format(zone.GetId(), os.path.basename(self.edbpath))
+            )
+            shutil.copytree(self.edbpath, edb_zone_path)
+            poly_data = zone.GetPolygonData()
+            edb_zones[edb_zone_path] = poly_data
+        return edb_zones
+
+    @pyaedt_function_handler()
+    def cutout_multizone_layout(self, zone_dict, common_reference_net):
+        """Create multizone project cutout.
+
+        Parameters
+        ----------
+        zone_dict : dict[str](EDB PolygonData)
+            Dictionary with EDB path as key and EDB PolygonData as value defining the zone region.
+            This dictionary is returned from the command copy_zones():
+            >>> edb = Edb(edb_file)
+            >>> zone_dict = edb.copy_zones(r"C:\Temp\test")
+
+        common_reference_net : str
+            the common reference net name. This net name must be provided to provide a valid project.
+
+        Returns
+        -------
+        dict[str][str] , dict[str][str]
+        first dictionary defined_ports with edb name as key and existing port name list as value
+        second dictionary terminals with edb name as key and created ports name on clipped signal nets
```

Comment:
```suggestion
        First dictionary defined_ports with the EDB name as the key and existing port name list as values.
        Second dictionary terminals with the EDB name as the key and created ports name on clipped signal nets.
```
I'm not sure how to suggest rewriting these descriptions, but they are rather choppy and more codelike than in plain English. For example, defined_ports as written seens like a parameter name that should be in double back ticks. However, we prefer simple descriptions that read like a sentence from a book.